client-docs/README.md

```diff
@@ -0,0 +1,363 @@
+    - [`$sync.setDiscovery`](#syncsetdiscovery)
+    - [`$sync.setSync`](#syncsetsync)
+    - [`$sync.stop`](#syncstop)
+    - Events
+      - [`'sync-state'`](#sync-state)
+  - [`project.$member`](#projectmember)
+    - [`$member.invite`](#memberinvite)
+    - [`$member.update`](#memberupdate)
+    - [`$member.remove`](#memberremove)
+    - Events
+      - [`'member-update'`](#member-update)
+  - [`project.$blob`](#projectblob)
+    - [`$blob.getUrl`](#blobgeturl)
+    - [`$blob.create`](#blobcreate)
+  - [DataType](#datatype)
+    - [Types](#types-2)
+    - [`dataType.create`](#datatypecreate)
+    - [`dataType.getByDocId`](#datatypegetbydocid)
+    - [`dataType.getByVersionId`](#datatypegetbyversionid)
+    - [`dataType.getMany`](#datatypegetmany)
+    - [`dataType.update`](#datatypeupdate)
+    - [`dataType.delete`](#datatypedelete)
+
+## Top Level Client
+
+Client interface for managing projects and invites. Provides an event emitter-like interface so it can emit and subscribe to events.
+
+### Types
+
+```ts
+import { Opaque } from "type-fest";
+
+type ProjectId = Opaque<string>;
+
+type ProjectInfo = {
+  id: ProjectId;
+  createdAt: Date;
+  updatedAt: Date;
+  name: string;
+};
+
+type ProjectRole = "creator" | "coordinator" | "member";
+
+type Invite = {
+  projectId: ProjectId;
+  projectName: string;
+  role: ProjectRole;
+};
+```
+
+### `client.createProject`
+
+`(opts: { name?: string }) => Promise<ProjectClient>`
+
+Create a new project instance. Resolves with a `ProjectClient` that exposes the [Project Client](#project-client) API.
+
+Accepts the following `opts`:
+
+- `name`: the name of the project
+
+### `client.getProject`
+
+`(projectId: string) => Promise<ProjectClient>`
+
+Retrieve a project client instance. Returns a `ProjectClient` that exposes the [Project Client](#project-client) API for the desired project.
+
+### `client.listProjects`
+
+`() => Promise<Array<ProjectInfo>>`
+
+Retrieve information about all projects.
+
+### `client.invite`
+
+Namespace for managing project invites. Provides an event emitter-like interface so it can emit and subscribe to events.
+
+#### `invite.accept`
+
+`(projectId: string) => Promise<ProjectClient>`
+
+Accept an invitation that was received for the project associated with `projectId`.
+
+#### `invite.reject`
+
+`(projectId: string) => Promise<void>`
+
+Reject an invitation that was received for the project associated with `projectId`.
+
+#### Events
+
+##### `'invite-received'`
+
+`invite.addEventListener('invite-received', (invite: Invite) => void)`
+
+Emits when a device invites you to a project.
+
+## Project Client
+
+Client interface for managing and interacting with a project. Built-in API methods and namespaces are prefixed with `$` to distinguish them from dynamic, application-specific namespaces that are created based on the [DataType](#datatype) API.
+
+### Types
+
+```ts
+import { Opaque } from "type-fest";
+
+type ProjectSettings = {
+  // TODO
+};
```

Comment:
Not sure what this looks like so needs some input

Reply:
Needs some more thinking, we can resolve this later